Cannot compile with GCC15

### System Information
TrenchBroom v2026.1-RC3

### Expected Behavior
Compilation should be possible with GCC15.

### Steps to Reproduce
Try to build the project with GCC15. It will fail with the same error as #4915, and once that specific file is patched, there are dozens more errors related to the same issue: 

```cmake
[...]
[ 21%] Building CXX object common/CMakeFiles/common.dir/src/mdl/Autosaver.cpp.o
In file included from /home/kinker31/QuakeMisc/TrenchBroom/common/src/mdl/Autosaver.cpp:31:
/home/kinker31/QuakeMisc/TrenchBroom/lib/kdl/src/kdl/ranges/enumerate_view.h:188:41: error: ‘const class kdl::ranges::enumerate_view<V>::iterator<Const>’ has no member named ‘current_’ [-Wtemplate-body]
  188 |       noexcept(std::ranges::iter_move(i.current_))
      |                                         ^~~~~~~~
make[3]: *** [common/CMakeFiles/common.dir/build.make:1325: common/CMakeFiles/common.dir/src/mdl/Autosaver.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:805: common/CMakeFiles/common.dir/all] Error 2
```

I was updating the Flatpak package to get it ready for this year's releases and came across this error, which was fixed by downgrading the runtime package on Flatpak's metadata to runtime v24, as the latest version, v25, provides GCC15 and that's not supported so far.

I also found out that the fix proposed in #4938 got "reverted" in https://github.com/TrenchBroom/TrenchBroom/pull/5222 (judging the fact that Claude was used for assistance, it may have ignored that GCC15 has stricter compile rules than its previous versions).

--- a/lib/KdLib/include/kd/ranges/enumerate_view.h
+++ b/lib/KdLib/include/kd/ranges/enumerate_view.h
@@ -188,7 +188,7 @@ public:
     }
 
     friend constexpr auto iter_move(const iterator& i) noexcept(
-      noexcept(std::ranges::iter_move(i.current_))
+      noexcept(std::ranges::iter_move(std::declval<iterator const&>().current_))
       and std::is_nothrow_move_constructible_v<
         std::ranges::range_rvalue_reference_t<Base>>)
     {

